pkg/frontend/admin_openshiftcluster_etcdcertificaterenew_test.go

```diff
@@ -0,0 +1,508 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/sirupsen/logrus"
+	"github.com/ugorji/go/codec"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/frontend/adminactions"
+	"github.com/Azure/ARO-RP/pkg/metrics/noop"
+	mock_adminactions "github.com/Azure/ARO-RP/pkg/util/mocks/adminactions"
+	utiltls "github.com/Azure/ARO-RP/pkg/util/tls"
+)
+
+func TestAdminEtcdCertificateRenew(t *testing.T) {
+	mockSubID := "00000000-0000-0000-0000-000000000000"
+	mockTenantID := "00000000-0000-0000-0000-000000000000"
+	mockRG := "resourceGroup"
+	ctx := context.Background()
+
+	type test struct {
+		name                 string
+		resourceID           string
+		version              *configv1.ClusterVersion
+		etcdoperator         *operatorv1.Etcd
+		etcdoperatorRevisied *operatorv1.Etcd
+		etcdCO               *configv1.ClusterOperator
+		notBefore            time.Time
+		notAfter             time.Time
+		mocks                func(*test, *mock_adminactions.MockKubeActions)
+		wantStatusCode       int
+		wantResponse         []byte
+		wantError            string
+	}
+
+	for _, tt := range []*test{
+		{
+			name:       "validate cluster version is <4.9",
+			resourceID: fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID),
+			version: &configv1.ClusterVersion{
+				Status: configv1.ClusterVersionStatus{
+					History: []configv1.UpdateHistory{
+						{
+							State:   configv1.CompletedUpdate,
+							Version: "4.11.44",
+						},
+					},
+				},
+			},
+			mocks: func(tt *test, k *mock_adminactions.MockKubeActions) {
+				k.EXPECT().
+					KubeGet(gomock.Any(), "ClusterVersion.config.openshift.io", "", "version").
+					Return(encodeClusterVersion(t, tt.version), nil)
+			},
+			wantStatusCode: http.StatusForbidden,
+			wantError:      "403: Forbidden: : etcd certificate renewal is not needed for cluster running version 4.9+",
```

Comment:
Nit: 403 usually indicates that an authenticated user doesn't have the right privileges to access the resource. I think a better status code would be 208 which means request successful but the resource was already processed.